### PR TITLE
Generate proper DWARF reg num for ARM32

### PR DESCRIPTION
**Pull request by kant2002**
After introduction of VFP-v3 ARM S0-S31 no longer can be generated using LLVM because  numbering of registers to start from 256 and only D0-D31 are used.
So this change encode S0 as D0, S2 as D1, etc. Also use reg nums for DXX registers.
This change fix generation of CFI codes,
which trigger issue with generation of DWARF using LLVM in NativeAOT
See https://developer.arm.com/documentation/ihi0040/c/?lang=en#dwarf-register-names
See https://github.com/dotnet/runtimelab/issues/1388

### CODE REVIEW

**Comment by JulieLeeMSFT** [comment]
@kunalspathak please review the community PR.
CC @dotnet/jit-contrib

**Review by jkotas — APPROVED** [review]
Thanks

**Comment by jkotas** [comment]
> please review the community PR.

Note that this is NativeAOT specific code currently.

File: src/coreclr/jit/unwind.cpp [hunk 1]
```diff
@@ -187,7 +187,12 @@ void Compiler::unwindPushPopMaskCFI(regMaskTP regMask, bool isFloat)
     regMaskTP regBit = isFloat ? genRegMask(REG_FP_FIRST) : 1;
 
     for (regNumber regNum = isFloat ? REG_FP_FIRST : REG_FIRST; regNum < REG_COUNT;
-         regNum           = REG_NEXT(regNum), regBit <<= 1)
+#if TARGET_ARM
```
Comment (kunalspathak) [inline src/coreclr/jit/unwind.cpp:190]: For better readability, I would rewrite this to.  I would add a short comment for why we need to do special case for ARM.

```c#
regNumber regNum = isFloat ? REG_FP_FIRST : REG_FIRST;
for (; regNum < REG_COUNT;)
{
#if TARGET_ARM
    regNum = isFloat ? REG_NEXT(REG_NEXT(regNum)) : REG_NEXT(regNum);
    regBit <<= isFloat ? 2 : 1;
#else
    regNum = REG_NEXT(regNum);
    regBit <<= 1;
#endregion
...
...
}

```

**Comment by kunalspathak** [comment]
LGTM - with 1 refactoring suggestion.

**Comment by kant2002** [comment]
@kunalspathak what suggestion? I did not see any of them.

**Review by kunalspathak — COMMENTED** [review]
This suggestion :) 
Forgot to publish it.

File: src/coreclr/jit/unwind.cpp (hunk 1 above)
Comment (kant2002) [reply src/coreclr/jit/unwind.cpp]: Done!

**Review by kunalspathak — APPROVED** [review]
LGTM, Thanks!

**Pull request merged**